### PR TITLE
 Refactor benchmark.go to remove `Dependencies` struct

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes

See #1138 for context.

This is based on the #1161 branch, so that PR needs to be merged first.

### Related Issues

- Blocked by: #1161
- Fixes: #1138 
- Related to #1093 

### Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: flavors/benchmark/eks.go [hunk 1]
```diff
@@ -74,25 +81,48 @@ func (k *EKS) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Conf
...
+
+func (k *EKS) checkDependencies() error {
+	missing := k.missingDependency()
+	if missing != "" {
+		return fmt.Errorf("%s is uninitialized", missing)
+	}
+	return nil
+}
+
+func (k *EKS) missingDependency() string {
```
Comment (uri-weisman) [inline flavors/benchmark/eks.go:111]: Returning an error instead of a string will make the code cleaner imo.

File: flavors/benchmark/gcp.go [hunk 2]
```diff
@@ -30,14 +31,21 @@ import (
...
-type GCP struct{}
+type GCP struct {
+	IdentityProvider identity.ProviderGetter
+}
 
 func (G *GCP) Run(context.Context) error { return nil }
 
-func (G *GCP) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo, dependencies *Dependencies) (registry.Registry, dataprovider.CommonDataProvider, error) {
-	gcpClientConfig, err := auth.GetGcpClientConfig(cfg, log)
+func (G *GCP) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo) (registry.Registry, dataprovider.CommonDataProvider, error) {
```
Comment (uri-weisman) [inline flavors/benchmark/gcp.go:43]:
```suggestion
func (g *GCP) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo) (registry.Registry, dataprovider.CommonDataProvider, error) {
```

File: flavors/posture.go [hunk 3]
```diff
@@ -71,15 +68,7 @@ func newPostureFromCfg(cfg *config.Config) (*posture, error) {
...
-	fetchersRegistry, cdp, err := b.Initialize(ctx, log, cfg, resourceCh, &benchmark.Dependencies{
-		AwsCfgProvider:           awslib.ConfigProvider{MetadataProvider: awslib.Ec2MetadataProvider{}},
-		AwsIdentityProvider:      awslib.IdentityProvider{},
-		AwsAccountProvider:       awslib.AccountProvider{},
-		GcpIdentityProvider:      identity.NewProvider(ctx, cfg, log),
-		KubernetesClientProvider: k8s.ClientGetter{},
-		AwsMetadataProvider:      awslib.Ec2MetadataProvider{},
-		EksClusterNameProvider:   awslib.EKSClusterNameProvider{},
-	})
+	fetchersRegistry, cdp, err := b.Initialize(ctx, log, cfg, resourceCh)
```
Comment (uri-weisman) [inline flavors/posture.go:71]: 🤩

File: flavors/benchmark/aws.go [hunk 4]
```diff
@@ -30,24 +31,25 @@ import (
...
 	"github.com/elastic/cloudbeat/resources/fetching/registry"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
 )
 
-type AWS struct{}
+type AWS struct {
+	IdentityProvider awslib.IdentityProviderGetter
+}
+
+func (A *AWS) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo) (registry.Registry, dataprovider.CommonDataProvider, error) {
```
Comment (uri-weisman) [inline flavors/benchmark/aws.go:41]: Should be changed in all methods
```suggestion
func (a *AWS) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Config, ch chan fetching.ResourceInfo) (registry.Registry, dataprovider.CommonDataProvider, error) {
```

File: flavors/benchmark/aws_org.go [hunk 5]
```diff
@@ -33,29 +34,31 @@ import (
 	"github.com/elastic/cloudbeat/resources/fetching"
 	"github.com/elastic/cloudbeat/resources/fetching/factory"
 	"github.com/elastic/cloudbeat/resources/fetching/registry"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
 )
 
-type AWSOrg struct{}
+type AWSOrg struct {
```
Comment (uri-weisman) [inline flavors/benchmark/aws_org.go:40]:
```suggestion
type AwsOrg struct {
```

**Review by uri-weisman — COMMENTED** [review]
Great work!
Commented mainly on minor things.

File: flavors/benchmark/aws_org.go (hunk 5 above)
Comment (orestisfl) [reply flavors/benchmark/aws_org.go:40]: I wonder what we should do for consistency with these names, the recommendation is to use all caps for acronyms: https://stackoverflow.com/a/63355758

Comment (orestisfl) [reply flavors/benchmark/aws_org.go:40]: Name was not really introduced in this PR so we can rename it on a different PR when we decide

Comment (uri-weisman) [reply flavors/benchmark/aws_org.go:40]: I see, so we can keep it the way it's.

File: flavors/benchmark/eks.go [hunk 6]
```diff
@@ -74,25 +81,40 @@ func (k *EKS) Initialize(ctx context.Context, log *logp.Logger, cfg *config.Conf
...
+		return fmt.Errorf("aws identity provider is uninitialized")
+	}
+	if k.ClientProvider == nil {
+		return fmt.Errorf("kubernetes client provider is uninitialized")
+	}
+	if k.EKSClusterNameProvider == nil {
+		return fmt.Errorf("EKS cluster name provider is uninitialized")
+	}
+	if k.AWSMetadataProvider == nil {
+		return fmt.Errorf("AWS metadata provideri is uninitialized")
```
Comment (uri-weisman) [inline flavors/benchmark/eks.go:117]:
```suggestion
		return fmt.Errorf("AWS metadata provider is uninitialized")
```

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**